AndroidSDKCore/src/main/java/com/leanplum/internal/Request.java

```diff
@@ -480,22 +480,88 @@ protected Void doInBackground(Void... params) {
     });
   }
 
-  private void sendRequests() {
+
+  /**
+   * This class wraps the unsent requests, requests that we need to send
+   * and the JSON encoded string. Wrapping it in the class allows us to
+   * retain consistency in the requests we are sending and the actual
+   * JSON string.
+   */
+  static class RequestsWithEncoding {
+    List<Map<String, Object>> unsentRequests;
+    List<Map<String, Object>> requestsToSend;
+    String jsonEncodedString;
+  }
+
+  private RequestsWithEncoding getRequestsWithEncodedStringForErrors() {
     List<Map<String, Object>> unsentRequests = new ArrayList<>();
     List<Map<String, Object>> requestsToSend;
+    String jsonEncodedRequestsToSend;
+
+    String uuid = UUID.randomUUID().toString();
+    for (Map<String, Object> error : localErrors) {
+      error.put(UUID_KEY, uuid);
+      unsentRequests.add(error);
+    }
+    requestsToSend = unsentRequests;
+    jsonEncodedRequestsToSend = jsonEncodeUnsentRequests(unsentRequests);
```

Comment:
I understand this is only for errors, but isn't there a risk that this `unsentRequests` also grows too big and crashes when `JsonConverter.toJson(data);` is called in line 726?

Reply:
Indeed. This whole process is wrapped in a try/catch block as in #176 so if errors is too large, we won't crash. Granted, we are not backing off there, but the error payload is very unlikely to be large. Event payload can grow since the param dictionary is unbounded.